### ISSUE TITLE
Dues redirect fields and delegates export for WFC

### ISSUE THREAD

**Issue by EdHollingdale** [thewca/worldcubeassociation.org#4923]
One of the items mentioned in #4199 was the ability to manage contacts for WCA Dues. While this is not a full automation of dues on the WCA website, it will make things easier for the WFC and add flexibility for Delegates. This feature request comes in two parts; adding additional fields to Delegate profiles, and adding an export option for WFC.

Context: one part of the Dues process is working out who to send the invoice to. By default this is sent to the first listed Delegate. However, some Delegates have redirects in place to send the invoice to an organisation/other individual (the Delegate is still ultimately responsible for paying the dues). For example, a typical redirect would be when a Delegate is the first listed Delegate in a particular country, the dues are sent to the regional organisation for that country. The current process means dues are automatically invoiced to the correct person, but the spreadsheet is incredibly manual to update for new Delegates or for changes in details.

### Additional Fields for Delegate Profiles
Ideally there would be an additional 4 fields on Delegate profiles editable by the Delegate and Senior: 

- Enable Redirect tickbox (if not enabled dues will just be emailed to the Delegate as standard). Explainer: “Would you like to enable a redirect for WCA Dues Invoices for competitions where you are the first listed delegate? Note that the Delegate is still ultimately responsible for ensuring payment of WCA Dues.”
- Redirect country dropdown (list of countries, including option for “All”). Explainer: “Which country would you like to enable the redirect for. Only competitions Delegated in this country will have invoices redirected”.
- Redirect Invoice Name. Explainer: “What is the name of the organization/individual the invoice should be issued to?”
- Redirect Email. Explainer: “What is the email address of the organization/individual the invoice should be issued to?”

For example, my redirect would be: Yes, Australia, Speedcubing Australia Inc, [email]

Like the Delegate location detail, this information should be hidden and excluded from developer exports etc.

### Delegates Export for WFC
Building on the fantastic work done in #4709 there would be an option for WFC members to download a csv/tsv of Delegates which would include Delegate Name, Delegate Email, and the four additional fields mentioned above.

Cc: @Jambrose777 as I know he has been working on unrelated work to update the Delegates table.

**Comment by Mollerz** [comment]
@viroulep  wfc tag pls

**Comment by viroulep** [comment]
As far as implementation is concerned, I think it would probably best to create a separate table for this (which also means no impact on the Delegates table).
This is something that will concerns (relatively) very very few of our users, so creating some kind of `has_one` relation between the users table and some new `due_redirect_for_users` seems appropriate.
Given country iso codes don't change, I think the field for the countries should be stored [serialized as an array](https://apidock.com/rails/ActiveRecord/AttributeMethods/Serialization/ClassMethods/serialize) of iso codes.

**Comment by Jambrose777** [comment]
Just a note: Would it be better for this to be on the competition level and not the user level?

Ie. I may have the following comps:
- Comp 1, where I want the dues sent to me, the Delegate
- Comp 2, where I was a supporting Delegate and the dues should be sent to the Main Delegate who handled finances.
- Comp 3, where the organizer handled finances
- Comp 4, where a regional organization handles finances.

You're suggestion would imply that I as a Delegate would always fall into 1 of the 4 categories, where it is often the case that it may be different per competition.

**Comment by EdHollingdale** [comment]
@Jambrose777 you're right, competition by competition would be good functionality - although we already have that in that Delegates can forward on an invoice (but are still responsible for paying it). However making it competition-by-competition would add extra steps for the people who are always in one category (about 1/3 of delegates). Currently those delegates don't have to do anything as it is captured in a spreadsheet the WFC maintains - this is really just shifting existing functionality to the website and making it user editable.

**Comment by Jambrose777** [comment]
> @Jambrose777 you're right, competition by competition would be good functionality - although we already have that in that Delegates can forward on an invoice (but are still responsible for paying it). However making it competition-by-competition would add extra steps for the people who are always in one category (about 1/3 of delegates). Currently those delegates don't have to do anything as it is captured in a spreadsheet the WFC maintains - this is really just shifting existing functionality to the website and making it user editable.

Gotchya, so both ideas can be implemented to cover both cases :) (I hadn't realized WFC was already doing this as a manual process with spreadsheets)
